### PR TITLE
74-transform-column-enhancement

### LINKED ISSUE #74: transform_column proposed improvement
`transform_column(df, col_name: str, function)` could be augmented to include a destination column. Example use case is when you want to perform a `log10` transformation, as in the docs example, yet you also want to preserve the non-transformed data for some other purpose.

### PR DESCRIPTION
**Pull request by zbarry**
This PR resolves #74 and my boredom on a Saturday.

# PR Checklist

Please ensure that you have done the following:

1. [X] PR in from a fork off your branch. Do not PR from <your_username>:master, but rather from <your_username>:<branch_name>.
2. [X] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [X] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checkin
    - running the test suite
    - docs build

## Code Changes

If you are adding code changes, please ensure the following:

- [X] Ensure that you have added tests.
- [X] Run all tests (`$ pytest .`) locally on your machine.
    - [X] Check to ensure that test coverage covers the lines of code that you have added.
    - [X] Ensure that all tests pass.

## Documentation Changes

If you are adding documentation changes, please ensure the following:

- [X] Build the docs locally.
- [X] View the docs to check that it renders correctly.

# PR Description

Please describe the changes proposed in the pull request:
- Gives user ability to store result of a column transformation in a new destination column

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

**Comment by zbarry** [comment]
I must have based this branch on my other PR... sigh.

**Comment by zbarry** [comment]
https://xkcd.com/1597/

File: tests/test_functions.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 def dataframe():
     data = {
         "a": [1, 2, 3] * 3,
-        "Bell__Chart": [1.23452345, 2.456234, 3.2346125] * 3,
+        "Bell__Chart": [1.234_523_45, 2.456_234, 3.234_612_5] * 3,
```
[the remark this answers is not in the document]
Comment (zbarry) [reply tests/test_functions.py]: Da eff

[the remark this answers is not in the document]
Comment (ericmjl) [reply tests/test_functions.py]: ??

[the remark this answers is not in the document]
Comment (zbarry) [reply tests/test_functions.py]: Wondering how those even got in there... I kept seeing them pop up when I was dealing with rebasing madness and thought I got rid of them.

**Comment by zbarry** [comment]
Uhhhh, turns out `make check` i.e., `black` is doing the underscore thing... wat.

**Comment by zbarry** [comment]
@ericmjl should be good to go.

**Comment by ericmjl** [comment]
> Uhhhh, turns out `make check` i.e., `black` is doing the underscore thing... wat.

Weird! Didn't think Black would do that. It's actually a good thing; makes the numbers readable. I have no strong opinions on this at the moment haha.

**Pull request merged**